Loop/Managers/LoopDataManager.swift

```diff
@@ -193,36 +197,49 @@ final class LoopDataManager: LoopSettingsAlerterDelegate {
     private var lockedSettings: Locked<LoopSettings>
 
     var settings: LoopSettings {
-        get {
-            lockedSettings.value
+        lockedSettings.value
+    }
+
+    @discardableResult func mutateSettings(_ changes: (_ settings: inout LoopSettings) -> Void) -> LoopSettings {
+        setSettingsWithResult({ (settings) -> LoopSettings in
+            changes(&settings)
+            return settings
+        })
+    }
```

Comment:
@ps2 What do you think about exposing the `mutateSettings` function? Exposing the needed properties of LoopSettings would still have threading issues, I believe.